test/parallel/test-stream-unpipe-event.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const {Writable, Readable} = require('stream');
+class NullWriteable extends Writable {
+  _write(chunk, encoding, callback) {
+    return callback();
+  }
+}
+class QuickEndReadable extends Readable {
+  _read() {
+    this.push(null);
+  }
+}
+class NeverEndReadable extends Readable {
+  _read() {}
+}
+
+const dest1 = new NullWriteable();
+dest1.on('pipe', common.mustCall(() => {}));
+dest1.on('unpipe', common.mustCall(() => {}));
+const src1 = new QuickEndReadable();
+src1.pipe(dest1);
+
+
+const dest2 = new NullWriteable();
+dest2.on('pipe', common.mustCall(() => {}));
+dest2.on('unpipe', () => {
```

Comment:
You could also use `common.mustNotCall('unpipe should not have been emitted')` in these cases.